xonsh/timings.py

```diff
@@ -223,3 +225,87 @@ def timeit_alias(args, stdin=None):
         if tc > tc_min:
             print("Compiler time: {0:.2f} s".format(tc))
     return
+
+# Can be used to insert custom timings into the code
+# events.on_timingprobe.fire(name='prope_test')
+# anywehere in the code base
+#@events.on_time_probe
+#def _time_prope_timing(name, **kw):
+#    if hasattr(builtins, '__xonsh_starttime__'):
+#        builtins.__xonsh_starttime__[name] = clock()
+##
+def setup_timings():
+    setattr(builtins, '__xonsh_timings__', {'startup': clock()})
```

Comment:
This initial call to clock() is not needed, since it should be assumed to be 0.0 to capture the interpreter startup time.